parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -287,4 +288,28 @@ public static boolean isIntType(PrimitiveType primitiveType) {
     }
     return primitiveType.getPrimitiveTypeName() == PrimitiveType.PrimitiveTypeName.INT32;
   }
+
+  public static String[] currentPath(Deque<String> fieldNames) {
```

Comment:
This implementation is too tied to how the `Deque` is used. I would expect iterating over the object passed in to create the right name, but this assumes it needs to use `descendingIterator()`. How about updating these methods to accept `Iterator<String>` instead so that they aren't tightly coupled to the original use?

Reply:
It's not appropriate  to put the two methods in ParquetUtil.java, I've moved them out of the utility class.  Thanks.

Reply:
That works for me.